python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -566,6 +566,19 @@ def has_assets_def(self) -> bool:
         assets_def = self.job_def.asset_layer.assets_def_for_node(self.node_handle)
         return assets_def is not None
 
+    @public
+    @property
+    def observable_asset_key(self) -> AssetKey:
```

Comment:
Any reason why we can't put this logic in the existing `asset_key` property?

Reply:
Just general trepidation for adjusting behavior of existing properties - I think the odds that someone is relying on `asset_key` raising an exception in this case is pretty much zero though so it's probably fine.

Reply:
IMO if we make `asset_key` work theres no need to add extra API surface